app/scripts/models/reliers/oauth.js

```diff
@@ -204,6 +207,42 @@ define(function (require, exports, module) {
     },
 
     /**
+     * Check if the relier wants access to the account encryption keys.
+     *
+     * @returns {Boolean}
+     */
+    wantsKeys () {
+      return !! (this.has('keysJwk') && this._config.scopedKeysEnabled && this._validateKeyScope());
+    },
+
+    /**
+     * Validate the requested scope with the relier redirect uri.
+     * At least one valid match must be found to successfully validate
+     * @returns {boolean}
+     * @private
+     */
+    _validateKeyScope () {
+      const validation = this._config.scopedKeysValidation || {};
+      let foundRedirectScopeMatch = false;
+
+      if (! this.get('scope')) {
+        throw new Error('Invalid scope parameter');
+      }
+
+      scopeStrToArray(this.get('scope')).forEach((scope) => {
+        if (validation.hasOwnProperty(scope)) {
+          if (validation[scope].redirectUris.includes(this.get('redirectUri'))) {
```

Comment:
Oh my, you can use new awesomeness because only new awesome browsers request keys!